### PR TITLE
NEWRELIC-3777 add docs for custom instrumentation in ESM

### PR DESCRIPTION
**Pull request by jmartin4563**
**What problems does this PR solve?**
* Creates documentation explaining how custom instrumentation works for ES module applications, including caveats about `newrelic.instrumentLoadedModule`

**Add any context that will help us review your changes such as testing notes,
  links to related docs, screenshots, etc.**
* Due to the nature of ES modules, we had to add a new configuration value that allows customers to inject their custom instrumentation definition into our ES module loader. This PR documents that configuration value, and also updates our newly created ESM documentation to link to our custom instrumentation example application. It also documents that `newrelic.instrumentLoadedModule` does not work in ESM apps, due to immutability of module bindings after load.

**If your issue relates to an existing GitHub issue, please link to it.**
* Part of NEWRELIC-3777

### CODE REVIEW

File: src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx [hunk 1]
```diff
@@ -1007,6 +1007,52 @@ This section defines the Node.js agent variables in the order they typically app
...
+          </th>
+
+          <td>
+            `NEW_RELIC_API_ESM_CUSTOM_INSTRUMENTATION_ENTRYPOINT`
+          </td>
+        </tr>
+      </tbody>
+    </table>
+
+    Complete filepath to custom instrumentation definition, enables the ability to load custom instrumentation in ES module applications.
```
Comment (bizob2828) [inline src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx:1049]: I think it's the relative path from the application root

File: src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx [hunk 2]
```diff
@@ -1007,6 +1007,52 @@ This section defines the Node.js agent variables in the order they typically app
...
+
+    Setting this configuration option will instruct the agent's ES module loader to load any custom instrumentation defined at the path provided.
+    To see a working example of defining custom instrumentation in an ES module application, check out our [example app](https://github.com/newrelic/newrelic-node-examples/tree/main/esm-app) in GitHub.
```
Comment (bizob2828) [inline src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx:1052]: do you think we should link directly to the custom entrypoint?

File: src/content/docs/apm/agents/nodejs-agent/installation-configuration/es-modules.mdx [hunk 3]
```diff
@@ -78,6 +78,13 @@ You can add the loader to your application by using the Node.js `--experimental-
 node --experimental-loader newrelic/esm-loader.mjs your-program.js
 ```
 
+## Custom instrumentation [#custom-instrumentation]
```
Comment (bizob2828) [inline src/content/docs/apm/agents/nodejs-agent/installation-configuration/es-modules.mdx:81]: I think you're going to want to update this [page](https://docs.newrelic.com/docs/apm/agents/nodejs-agent/api-guides/guide-using-nodejs-agent-api/#custom-instrumentation) and perhaps cross link the ESM section from here

File: src/content/docs/apm/agents/nodejs-agent/installation-configuration/es-modules.mdx [hunk 4]
```diff
@@ -78,6 +78,13 @@ You can add the loader to your application by using the Node.js `--experimental-
...
+The Node.js APM agent supports adding custom instrumentation to your ES module applications.
+You can create instrumentation in ES module applications using most of the [instrumentation registration methods on the API](/docs/apm/agents/nodejs-agent/api-guides/nodejs-agent-api/#custom-instrumentation-api).
+Due to the nature of ES module imports, we are unable to support `newrelic.instrumentLoadedModule`. ES module bindings are not assignable, which prevent the agent from replacing a module's exported members with instrumentation after the module has been loaded.
+
+To see a demonstration of how to use the custom instrumentation API in an ES module application, check out our [example app](https://github.com/newrelic/newrelic-node-examples/tree/main/esm-app) in GitHub.
```
Comment (bizob2828) [inline src/content/docs/apm/agents/nodejs-agent/installation-configuration/es-modules.mdx:86]: do we also want to put the link to this app at the bottom of the page with the other links?

File: src/content/docs/apm/agents/nodejs-agent/api-guides/nodejs-agent-api.mdx [hunk 5]
```diff
@@ -176,6 +176,10 @@ Use these API calls to [expand your instrumentation with custom instrumentation]
     id="instrumentLoadedModule"
     title={<InlineCode>newrelic.instrumentLoadedModule(moduleName, moduleInstance)</InlineCode>}
   >
+    <Callout variant="important">
+      This method is not supported in ES module applications.
```
Comment (bizob2828) [inline src/content/docs/apm/agents/nodejs-agent/api-guides/nodejs-agent-api.mdx:180]: Not sure if we want to call this out but it's technically not needed in ESM. Maybe I'm wrong but due to how the experimental loader registers instrumentation I don't think we have the same issues we did with bootstrapping agent in cjs

**Review by bizob2828 — CHANGES_REQUESTED** [review]
Nice work. a few comments/questions

File: src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx (hunk 1 above)
Comment (jmartin4563) [reply src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx:1049]: We're using `path.resolve` which supports full and relative filepaths, right? Do you think I should change this to be something like: ```Filepath to custom instrumentation definition, which can be absolute or relative to the application root. This enables the ability...```

Comment (bizob2828) [reply src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx:1049]: realistically they won't be able to provide a full path due to the nature of apps starting. I would just say relative to where the application is started

Comment (jmartin4563) [reply src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx:1049]: Wouldn't they be able to provide a full file path if they're using an env var? I'll change it to say relative, but flagging it as a potential source of confusion, which we can cross that bridge when we get to it.

**Review by bizob2828 — APPROVED** [review]

File: src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx [hunk 6]
```diff
@@ -1007,6 +1007,52 @@ This section defines the Node.js agent variables in the order they typically app
...
+          </th>
+
+          <td>
+            `NEW_RELIC_API_ESM_CUSTOM_INSTRUMENTATION_ENTRYPOINT`
+          </td>
+        </tr>
+      </tbody>
+    </table>
+
+    Relative filepath to custom instrumentation definition, which is relative to the application root. Enables the ability to load custom instrumentation in ES module applications.
```
Comment (rhetoric101) [inline src/content/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration.mdx:1049]: @jmartin4563, how about a minor revision like this: 

This uses the relative filepath to the custom instrumentation definition, which is relative to the application root. It enables you to load custom instrumentation in ES module applications.

**Review by rhetoric101 — APPROVED** [review]
@jmartin4563 I just made a few suggestions in one paragraph. Take a look and see what you think.

**Review by bizob2828 — APPROVED** [review]

**Review by rhetoric101 — APPROVED** [review]
@jmartin4563 this looks good!

**Pull request merged**